### DISCUSSION TITLE
Should we implement more batch create functions?

### DISCUSSION THREAD

**Discussion by andreivladbrg** [sablier-labs/v2-periphery#20] [General]
In the current version of the target contract I've implemented the following batch create functions:
- `batchCreateWithDurations`
- `batchCreateWithRange`
- `batchCreateWithDeltas`
- `batchCreateWithMilestones`

Should we also implement functions such as `batchCreateWithDurationsAndDeltas` and `batchCreateWithRangeAndMilestones`?

**Comment by PaulRBerg** [comment]
Thanks for opening this discussion, and sorry for taking so long to reply.

As I have argued in #13, we should apply the YAGNI principle, which means not implementing logic we don't know for sure that will be needed.

The UI is not currently designed to create groups of streams across different streaming contracts (i.e. `LockupLinear` and `LockupDynamic`), so there is no need to implement any other create functions in the near future.
